src/executer.ts

```diff
@@ -0,0 +1,57 @@
+import { TaskQueue } from './types'
+
+const QUEUE = 'commandQueue'
+const AGENTS = 'agents'
+
+export default class Executer {
+  private createAgent: Function
+  private commandQueue: any[]
+  private agents: any
+
+  constructor (Agent: any) {
+    this.createAgent = () => new Agent()
+    this.commandQueue = []
+    this.agents = {}
+  }
+
+  run (args: any) {
+    let cmd: TaskQueue[] = this.parse(args)
+    cmd = this.commandQueue.concat(cmd)
+    for (this[QUEUE] = []; cmd.length > 0;) {
+      if (this.execute(cmd[0])) {
+        cmd.shift()
+      } else {
+        break
```

Comment:
why break

Reply:
To stop queue. If give invalid queue command, will be crash in `execute`.

Reply:
crash test is [here](https://github.com/userdive/agent.js/pull/167/files#diff-c7c32dc986af3dddacf3be3fc5cce2a1R44).